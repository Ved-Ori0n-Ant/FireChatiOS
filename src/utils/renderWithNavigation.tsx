import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

const renderWithNavigation = (renderComponent: any) => {
  return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
}

export default renderWithNavigation;
