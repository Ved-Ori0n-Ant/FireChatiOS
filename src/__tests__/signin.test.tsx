import React from "react";
import { cleanup, render, fireEvent, screen } from '@testing-library/react-native'
import HomeScreen from "../screens/signedInModule/signInScreen";
import '@testing-library/jest-native/extend-expect';
import auth from '@react-native-firebase/auth';
import { NavigationContainer } from "@react-navigation/native";
import renderWithNavigation from "../utils/renderWithNavigation";
// import firebase from '@react-native-firebase/app'

afterEach(cleanup);

jest.mock('@react-native-firebase/auth', () => ({
  signInWithEmailAndPassword: jest.fn()
}));

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    dispatch: mockedNavigate,
  }),
}));

describe('Tests for signin screen', ()=>{
  it('should render correctly', () => {
    render(
      <NavigationContainer>
        <HomeScreen />
      </NavigationContainer>
    );
    const tree: any = screen;
    expect(tree.toJSON).toMatchSnapshot();
  });
  it('should have text', () => {
    // const tree = render(
    //   <NavigationContainer>
    //     <HomeScreen />
    //   </NavigationContainer>
    // )
    const tree = renderWithNavigation(<HomeScreen />)
    const defaultText: any = tree.queryByTestId('const-text');
    expect(defaultText.props.children).toBe('Please enter your credentials:')
  })
})

it('placeholder default values', () => {
  // render(<HomeScreen />);
  const tree = renderWithNavigation(<HomeScreen />)
  const emailInput = tree.getByTestId('email-input');
  const passwordInput = tree.getByTestId('password-input');
  const submitBtn = tree.getByTestId('signin-btn');
  // const logInPressed = jest.fn();

  fireEvent.changeText(emailInput, 'abc@def.ghi')
  fireEvent.changeText(passwordInput, '121212')
  fireEvent.press(submitBtn)

  expect('abc@def.ghi').toBeDefined()
  expect('121212').toBeDefined()
  // expect(logInPressed).toBeCalled()
  expect(passwordInput.props.value).toBe('121212')
  expect(emailInput.props.value).toBe('abc@def.ghi')
})

it('All touchables should be defined', () => {
  // const tree = render(
  //   <NavigationContainer>
  //     <HomeScreen/>
  //   </NavigationContainer>
  // )
  const tree = renderWithNavigation(<HomeScreen />)
  const signinBtn = tree.getByTestId('signin-btn');
  const signinPhoneBtn = tree.getByTestId('signin_with_phone')
  const signUpBtn = tree.getByTestId('sign_up_btn')
  const signinBtnStatus = signinBtn.props.accessibilityState.disabled;
  const signinPhoneBtnStatus = signinPhoneBtn.props.accessibilityState.disabled;
  const signUpBtnStatus = signUpBtn.props.accessibilityState.disabled;

  expect(signinBtn).not.toBeNull();
  expect(signinPhoneBtn).not.toBeNull();
  expect(signUpBtn).not.toBeNull();
  expect(signinBtnStatus).toBe(false);
  expect(signinPhoneBtnStatus).toBe(false);
  expect(signUpBtnStatus).toBe(false);
})