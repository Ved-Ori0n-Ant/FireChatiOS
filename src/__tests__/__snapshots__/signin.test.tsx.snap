// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for signin screen should render correctly 1`] = `[Function]`;
